Kernel/Tasks/PowerStateSwitchTask.h

```diff
@@ -4,8 +4,39 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#pragma once
+
+#include <AK/Forward.h>
+#include <Kernel/Forward.h>
+
 namespace Kernel {
 
+enum class PowerStateCommand : uintptr_t {
+    Shutdown,
+    Reboot,
+};
+// We will pass the power state command to the task in place of a void* as to avoid the complications of raw allocations.
+static_assert(sizeof(PowerStateCommand) == sizeof(void*));
```

Comment:
Apologies, either this is new or I just missed this last time: what "complications of raw allocations" are there with this enum? `spawn` goes ahead and creates a `KString` and a new process, not sure what this is trying to achieve?

Reply:
Avoid an allocation, and ownership responsibilities. This just makes things simpler.

Reply:
Ah the ownership POV makes sense to me - gotcha.